src/dmm_api.cpp

```diff
@@ -189,4 +189,20 @@ void DMM_API::setNotes(QString str)
 	dmm->ui->instrumentNotes->setNotes(str);
 }
 
+QVector<int> DMM_API::getGainModes() const
+{
+	return {dmm->ui->gainCh1ComboBox->currentIndex(),
+		dmm->ui->gainCh2ComboBox->currentIndex()};
+}
+
+void DMM_API::setGainModes(const QVector<int> &gainModes)
+{
+	if (gainModes.size() < 2) {
```

Comment:
We can also use **m_adc_nb_channels** here, instead of 2..right?

Reply:
Yup we can use it here, as for each channel a gain control is associated. I think this can be done in a better way by providing a `DMMChannelWidget`  thus having to instantiate one for each channel (in `m_adc_nb_channels`) instead of duplicating everything. Each having an API with individual controls, but that's not the purpose of this pr.